src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -55,7 +56,7 @@ public function process(ContainerBuilder $container)
 
             $commandId = 'console.command.'.strtolower(str_replace('\\', '_', $class));
 
-            if (!isset($tags[0]['command'])) {
+            if (!isset($tags[0]['command']) && !$r->implementsInterface(DefaultNameProviderInterface::class)) {
```

Comment:
Hm right. Would application as a service be the next step? So we can imply wire via add method call.